bundled/muparser_v2_3_3/src/muParserTest.cpp

```diff
@@ -1546,17 +1546,7 @@ namespace mu
 					// The tests equations never result in infinity, if they do thats a bug.
 					// reference:
 					// http://sourceforge.net/projects/muparser/forums/forum/462843/topic/5037825
-#ifdef _MSC_VER
-#pragma warning(push)
-#pragma warning(disable:4127)
-#endif
-					if (std::numeric_limits<value_type>::has_infinity)
-#ifdef _MSC_VER
-#pragma warning(pop)
-#endif
-					{
-						bCloseEnough &= (fabs(fVal[i]) != numeric_limits<value_type>::infinity());
-					}
+					bCloseEnough &= isfinite(fVal[i]);
```

Comment:
The C version of `isfinite` has interesting semantics regarding the argument's type. You probably want to use `std::isfinite` instead.

Reply:
They actually use the std namespace here.
https://github.com/dealii/dealii/blob/d3c7782fdc2eb4621528270212289177561e14e4/bundled/muparser_v2_3_3/src/muParserTest.cpp#L36

But we can be more explicit with std::isfinite

Reply:
Oh, I see. Either is fine then. If this was in deal.II, we'd know what to do but here it perhaps doesn't matter.

Reply:
I moved it to `std::isfinite` in both patches to remove ambiguity.

@bangerth -- Could you approve the patch then :)